jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c

```diff
@@ -149,10 +148,10 @@ ecma_builtin_binary_floating_number_to_string (lit_utf8_byte_t *digits_p, /**< n
 
   if (to_num_digits > 0)
   {
-    /* Add significant digits of the fraction part. */
+    /* Add significant digits of the fraction part and fill the remaining digits with zero */
     while (to_num_digits > 0)
     {
-      *p++ = num_digits == 1 ? '0' : *digits_p++;
```

Comment:
This still won't solve the issue, we're still reading from outside of the `digits_p` buffer, and it's only by sheer chance that we're seeing zeros there (It's a UTF string, it's not zero terminated). Instead, we need to check how many digits we need (`to_num_digits`) versus how many digits we actually have in the buffer (`num_digits`), and act accordingly if we run out.

Reply:
Okay, after checking, we _do_ end up with a zero terminated string in `digits_p`, disregard my previous comment.

Reply:
Exactly :) Thanks for your review!